### PR TITLE
Feature/receiving post request

### PR DESCRIPTION
**Pull request by dhee-tree**
### What

- Changed configs for sending email
- Upgraded dependencies, go version and linter
- Updated test format to match configs
- Update component test

### How to review

- CI test pass
- Sense check

### Who can review

!Me

### CODE REVIEW

File: ci/component.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.19.4
+    tag: 1.23.1-bullseye
```
Comment (lindenmckenzie) [inline ci/component.yml]:
```suggestion
    tag: 1.23.2-bullseye
```

Comment (lindenmckenzie) [reply ci/component.yml]: Same for others.

File: config/config.go [hunk 2]
```diff
@@ -22,10 +22,10 @@ type Config struct {
...
 // Mail represents the subset of configuration corresponding to the email service
 type Mail struct {
-	Host     string `envconfig:"MAIL_HOST"`
-	User     string `envconfig:"MAIL_USER"`
-	Password string `envconfig:"MAIL_PASSWORD" json:"-"`
-	Port     string `envconfig:"MAIL_PORT"`
+	MailHost     string `envconfig:"MAIL_HOST"`
+	MailUser     string `envconfig:"MAIL_USER"`
+	MailPassword string `envconfig:"MAIL_PASSWORD" json:"-"`
+	MailPort     string `envconfig:"MAIL_PORT"`
```
Comment (lindenmckenzie) [inline config/config.go]: Is this necessary? Was the linter complaining? Otherwise it's Mail.MailHost which seems a bit wordy.

File: features/feedback.feature [hunk 3]
```diff
@@ -21,8 +21,7 @@ Feature: Feedback
         To: receiver@feedback.com
         Subject: Feedback received
 
-        Is page useful: true
-        Is general feedback: false
+        Feedback Type: A specific page
         Page URL: https://localhost/subpath/one
         Description: very nice and useful website!
```
Comment (lindenmckenzie) [inline features/feedback.feature:26]: Do we have a test for the 'is general feedback: true' path?

File: go.mod [hunk 4]
```diff
@@ -1,6 +1,6 @@
 module github.com/ONSdigital/dp-feedback-api
 
-go 1.19
+go 1.23
 
 replace google.golang.org/grpc => google.golang.org/grpc v1.55.0
```
Comment (lindenmckenzie) [inline go.mod]: These replace statements should be removed as per our standard approach to dependency management.

File: main_test.go [hunk 5]
```diff
@@ -23,14 +23,6 @@ func (f *ComponentTest) InitializeScenario(ctx *godog.ScenarioContext) {
...
 
-	ctx.BeforeScenario(func(*godog.Scenario) {
-		component.Reset()
-	})
-
-	ctx.AfterScenario(func(*godog.Scenario, error) {
-		_ = component.Close()
-	})
-
 	component.RegisterSteps(ctx)
```
Comment (lindenmckenzie) [inline main_test.go]: Any reason this has been removed?

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]
Few little bits.

File: api/feedback.go [hunk 6]
```diff
@@ -43,24 +46,27 @@ func GenerateFeedbackMessage(f *models.Feedback, from, to string) []byte {
 
 	b.WriteString(fmt.Sprintf("From: %s\n", from))
 	b.WriteString(fmt.Sprintf("To: %s\n", to))
-	b.WriteString(fmt.Sprintf("Subject: Feedback received\n\n"))
+	b.WriteString("Subject: Feedback received\n\n")
 
-	b.WriteString(fmt.Sprintf("Is page useful: %t\n", *f.IsPageUseful))
-	b.WriteString(fmt.Sprintf("Is general feedback: %t\n", *f.IsGeneralFeedback))
+	if *f.IsPageUseful {
+		b.WriteString(fmt.Sprintf("Feedback Type: %s\n", mapper.ASpecificPage))
```
Comment (lindenmckenzie) [inline api/feedback.go]:
```suggestion
	if !*f.IsGeneralFeedback {
		b.WriteString(fmt.Sprintf("Feedback Type: %s\n", mapper.ASpecificPage))
```

Comment (lindenmckenzie) [reply api/feedback.go]: or you could even just provide the string based off that, saves a minor point of duplication.

File: api/feedback_test.go [hunk 7]
```diff
@@ -38,8 +25,7 @@ var expectedEmail = `From: sender@mail.com
 To: receiver@mail.com
 Subject: Feedback received
 
-Is page useful: true
-Is general feedback: false
+Feedback Type: A specific page
 Page URL: https://testhost:1234/sub/path
 Description: very nice and useful website!
 Name: Mr Feedback reporter
```
Comment (lindenmckenzie) [inline api/feedback_test.go:32]: we could expand the testing here to get both cases.

**Review by lindenmckenzie — COMMENTED** [review]
Some more minor comments.

File: ci/component.yml (hunk 1 above)
Comment (dhee-tree) [reply ci/component.yml]: Done

File: config/config.go (hunk 2 above)
Comment (dhee-tree) [reply config/config.go]: Done

File: api/feedback_test.go [hunk 8]
```diff
@@ -30,23 +17,44 @@ var (
...
 )
 
 var expectedEmail = `From: sender@mail.com
 To: receiver@mail.com
 Subject: Feedback received
 
-Is page useful: true
-Is general feedback: false
+Feedback Type: A specific page
+Page URL: https://testhost:1234/sub/path
```
Comment (lindenmckenzie) [inline api/feedback_test.go:30]: Should we be printing the URL if it's general feedback? Maybe these should be mutually exclusive?

**Review by lindenmckenzie — COMMENTED** [review]
Tiny comment to have a think about.

File: go.mod (hunk 4 above)
Comment (dhee-tree) [reply go.mod]: Done

**Review by lindenmckenzie — APPROVED** [review]
